src/codegen/utilities/interpreter_helpers.py

```diff
@@ -51,6 +53,37 @@ def get_interpreter_functions(metadata):
     return sorted(functions_metadata, key=lambda x: x._function_name)
 
 
+def is_grpc_only_function(function_data):
+    """Returns true if the function is a grpc only function."""
+    if "codegen_method" in function_data:
+        if function_data["codegen_method"] == "grpc-only":
+            return True
+
+
+def generate_interpreter_function_call_args(function_metadata):
+    """Gets function call arguments."""
+    function_call_args = []
+    # if function_metadata.handle_parameter is not None:
+    #     if function_metadata.handle_parameter.cvi_name == "taskHandle":
+    #         function_call_args.append("task")
+    for param in function_metadata.base_parameters:
+        if param.direction == "in":
+            function_call_args.append(param.parameter_name)
+            if param.has_explicit_buffer_size:
+                function_call_args.append(f"len({param.parameter_name})")
+        else:
+            if param.has_explicit_buffer_size:
+                function_call_args.append(param.parameter_name)
+                function_call_args.append("temp_size")
+            else:
+                function_call_args.append(f"ctypes.byref({param.parameter_name})")
+
+    if function_metadata.calling_convention == "Cdecl":
+        function_call_args.append("None")
```

Comment:
This is sometimes correct but it's not correct in the general case.

In general, the DAQmx C API uses `__cdecl` for varargs functions (with a few exceptions). There are a few common patterns for designing varargs functions:
- [`printf`-style](https://man7.org/linux/man-pages/man3/printf.3.html): a format string parameter specifies how to unpack the variable arguments
- [`exec`-style](https://man7.org/linux/man-pages/man3/exec.3.html): the variable arguments specify array elements, with a null parameter to terminate the array.
- [`ioctl`-style](https://man7.org/linux/man-pages/man3/ioctl.3p.html): a numeric parameter specifies which operation to execute and how to interpret the variable arguments.

Most DAQmx programmable power-up state functions are execl-style. These need a null argument as a terminator. (At some point, we started using arrays instead of varargs, to make it easier to choose the number of arguments at run time. I guess that would be execv-style.)

Generic attribute get/set functions like `DAQmxGetChanAttribute`, `DAQmxSetChanAttribute` are ioctl-style. They do not need a null argument as a terminator.

Reply:
Thanks for the information, based on this we have decided to take up the logic for handling cdecl calls in a separate PR. Currently this is being tracked by [Task 2357886](https://ni.visualstudio.com/DevCentral/_workitems/edit/2357886): Handle the cdecl calls in library interpreter